open_feature/open_feature_client.py

```diff
@@ -280,3 +280,6 @@ def _create_provider_evaluation(
             raise GeneralError(error_message="Unknown flag type")
 
         return get_details_callable(*args)
+
+    def set_evaluation_context(self, evaluation_context: EvaluationContext):
```

Comment:
Is there a specific reason we need this setter method? Feels like a bit of a python anti-pattern since one could just set it directly on the object? We also already have the evaluation context in the constructor so I'm unclear what this method gives us? 

If there's a good reason to keep it, however, I'm keen to structure the methods slightly, primarily just keeping public methods above any private methods.

Reply:
The typical way a client is created is not though the constructor but through the `openFeatureAPI.get_client()` method. I agree its not necessary though and we can just set the value directly, unless we want it to be more similar to other sdks. I'm happy to remove it though.

Reply:
Ah yes, that's a good point. I see that you removed it already - lets leave it out and then if we find that it would be useful to have in the future we can add it in at a later stage. Thanks!